### PR TITLE
Fix batch size calculation with vmap and static_argnums

### PR DESCRIPTION
**Pull request by dime10**
Resolves a bug accessing argument shapes in the vmap function when those arguments don't have a shape. This might happen when using vmap together with `static_argnums` for example:

```py
@qjit(static_argnums=0)
@vmap(in_axes=(None, 0))
def test(n, array):
    for _ in range(n):
        array = array * 2
    return array

test(10, np.random.rand(3,2))
```

Also improves error message for out of bounds axis specifier.

[sc-74291]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -187,6 +187,9 @@
 
 <h3>Bug fixes</h3>
 
+* Resolve a bug in the `vmap` function when passing shape-less values to the target.
```
Comment (joeycarter) [inline doc/releases/changelog-dev.md:190]:
```suggestion
* Resolve a bug in the `vmap` function when passing shapeless values to the target. 
```

File: frontend/test/pytest/test_vmap.py [hunk 2]
```diff
@@ -782,3 +784,24 @@ def fn(x):
...
+
+        @vmap(in_axes=(None, 0))
+        def f(n, arr):
+            return n * arr
+
+        batch_size = f._get_batch_size((arg, jnp.zeros((2, 5))), (None, 0), None)
+        assert batch_size == 2
+
+    def test_vmap_axis_out_of_bounds(self):
+        """Confirm that vmap works when used with shape-less values."""
```
Comment (joeycarter) [inline frontend/test/pytest/test_vmap.py:800]:
```suggestion
        """Confirm that vmap works when used with shapeless values."""
```

**Review by joeycarter — APPROVED** [review]
